src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserStateObject.Multiplexer.cs

```diff
@@ -0,0 +1,559 @@
+                        bytesConsumed += headerBytesAvailable;
+                        data = data.Slice(headerBytesAvailable);
+                    }
+                    if (partialPacket.HasHeader)
+                    {
+                        partialPacket.DataLength = Packet.GetDataLengthFromHeader(partialPacket);
+                    }
+                }
+
+                if (partialPacket.HasDataLength)
+                {
+                    if (partialPacket.CurrentLength < partialPacket.RequiredLength)
+                    {
+                        // the packet length is known so take as much data as possible from the incoming
+                        // data to try and complete the packet
+
+                        int payloadBytesNeeded = partialPacket.DataLength - (partialPacket.CurrentLength - TdsEnums.HEADER_LEN);
+                        int payloadBytesAvailable = Math.Min(data.Length, payloadBytesNeeded);
+
+                        ReadOnlySpan<byte> payloadSource = data.Slice(0, payloadBytesAvailable);
+                        Span<byte> payloadTarget = partialPacket.Buffer.AsSpan(partialPacket.CurrentLength, payloadBytesAvailable);
+                        payloadSource.CopyTo(payloadTarget);
+
+                        partialPacket.CurrentLength = partialPacket.CurrentLength + payloadBytesAvailable;
+                        bytesConsumed += payloadBytesAvailable;
+                        data = data.Slice(payloadBytesAvailable);
+                    }
+                    else if (partialPacket.CurrentLength > partialPacket.RequiredLength)
+                    {
+                        // the partial packet contains a complete packet of data and also contains
+                        // data from a following packet
+
+                        // the TDS spec requires that all packets be of the defined packet size apart from
+                        // the last packet of a response. This means that is should not possible to have more than
+                        // 2 packet fragments in a single buffer like this:
+                        //  - first packet caused the partial
+                        //  - second packet is the one we have just unpacked
+                        //  - third packet is the extra data we have found
+                        // however, due to the timing of cancellation it is possible that a response token stream
+                        // has ended before an attention message response is sent leaving us with a short final
+                        // packet and an additional short cancel packet following it
+
+                        // this should only happen when the caller is trying to consume the partial packet
+                        // and does not have new input data
+
+                        int remainderLength = partialPacket.CurrentLength - partialPacket.RequiredLength;
+
+                        partialPacket.CurrentLength = partialPacket.CurrentLength;
```

Comment:
Did you mean?
```suggestion
                        partialPacket.CurrentLength = partialPacket.RequiredLength;
```

Reply:
fixed. but the ci doesn't seem to be running.